src/main/java/com/mg105/entities/Inventory.java

```diff
@@ -84,8 +86,42 @@ public boolean removeItem(String itemName) {
         return false;
     }
 
+    /**
+     * Uses the item on the given character
+     *
+     * @param character The character to use the item on
+     * @param itemName  The name of the item to use
+     * @return true iff the item with a name of itemName was used
+     * @see Item
+     * @see Consumable
+     */
+
+    public boolean useItem(BattleCharacter character, String itemName) {
+        for (int i = 0; i < numberOfItems(); i++) {
+            Item item = this.items.get(i);
+
+            if (item.getName().equals(itemName)) {
+
+                if (item instanceof Consumable consumableItem) {
+                    consumableItem.consume(character);
+                    this.items.remove(i);
+                    return true;
+
+                } else {
+
+                    return false;
```

Comment:
I don't think this else statement is needed since there is already a return false after the loop, but I guess it can reduce the number of loop iterations in some cases.